### PR TITLE
Remove rgdal

### LINKED ISSUE #125: rgdal was retired from CRAN and is required for Processing R Provider
Hi,
I'm trying to get Processing R Provider, however it seems to require rgdal in order to work. 
in the output, i get this message:

```
Installing package into 'C:/Users/idshk/AppData/Roaming/QGIS/QGIS3/profiles/default/processing/rlibs'
(as 'lib' is unspecified)
Warning message:
package 'rgdal' is not available for this version of R

A version of this package for your version of R might be available elsewhere,
see the ideas at
https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages
Error in library("rgdal") : there is no package called 'rgdal'
Execution halted
```

### PR DESCRIPTION
**Pull request by JanCaha**
This is rather large change in the code, that reflects that R package `rgdal` has been retired from CRAN. Thus parts of the plugin are not usable anymore and actually cause issue with trying to install non existing package.

This should not be large problem, as this was quite heavily discussed in R community and the package was kind of obsolete anyways. Still this is quite likely to render some old scripts unusable and with need to be updated. 

Some users would be able to run their old scripts if they have currently installed `rgdal`, but they will loose this option when upgrading their R version or reinstalling. For compatibility, older plugin versions can be used. But I believe that the latest version of the plugin should only support currently working R setup (which does not have `rgdal` at all).

@nyalldawson any issues with this?  If not, could you please merge this and release the version to QGIS repository?

Fixes #125

### CODE REVIEW

File: processing_r/processing/algorithm.py [hunk 1]
```diff
@@ -291,16 +291,6 @@ def process_metadata_line(self, line):  # pylint: disable=too-many-return-statem
                                                                     self.tr('HTML files (*.html)'), optional=True))
             return
 
-        # dontuserasterpackage is the older version, should be considere obsolete
-        if line.lower().strip().startswith('load_raster_using_rgdal') or \
```
Comment (nyalldawson) [inline processing_r/processing/algorithm.py:295]: Should we raise a QgsProcessingException with a nice explanatory message if this line is found?

**Comment by nyalldawson** [comment]
Thanks for doing this @JanCaha -- this is a fantastic job!

File: processing_r/processing/algorithm.py (hunk 1 above)
Comment (JanCaha) [reply processing_r/processing/algorithm.py:295]: True. Fixed.

**Comment by JanCaha** [comment]
Added the QgsProcessingError if the unsupported commands are found,  so it should be ready for release. @nyalldawson

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
@JanCaha can I make you an admin for this plugin in the qgis repo? It seems silly for you not to have the ability to publish releases 😂

**Comment by JanCaha** [comment]
@nyalldawson feel free to do so :) User name is **cahik** at the plugins.qgis.org

**Comment by JanCaha** [comment]
Did and also pushed the release to QGIS plugins repository :)

**Comment by nyalldawson** [comment]
Thank you!!

**Pull request merged**